src/sessions/p2p_session.rs

```diff
@@ -865,46 +864,59 @@ impl<T: Config> P2PSession<T> {
 
     fn compare_local_checksums_against_peers(&mut self) {
         let current = self.current_frame();
-        if current % CHECKSUM_REPORT_INTERVAL == 0 && current > self.max_prediction as i32 {
-            for (_, remote) in &self.player_reg.remotes {
-                for (remote_frame, remote_checksum) in remote.checksum_history() {
-                    for (local_frame, local_checksum) in &self.checksum_history {
-                        // if checksums are equal for the same frame send desync event
-                        if *local_frame == *remote_frame && *local_checksum != *remote_checksum {
-                            self.event_queue.push_back(GGRSEvent::DesyncDetected {
-                                frame: *local_frame,
-                                local_checksum: *local_checksum,
-                                remote_checksum: *remote_checksum,
-                                remote_addr: remote.peer_addr(),
-                            });
-                            // exit function desync already detected
-                            return;
+        match self.desync_detection {
+            DesyncDetection::On { interval } => {
+                if current % interval as i32 == 0 {
+                    for (_, remote) in &self.player_reg.remotes {
+                        for (remote_frame, remote_checksum) in remote.checksum_history() {
+                            for (local_frame, local_checksum) in &self.checksum_history {
+                                // if checksums are equal for the same frame send desync event
+                                if *local_frame == *remote_frame
+                                    && *local_checksum != *remote_checksum
+                                {
+                                    self.event_queue.push_back(GGRSEvent::DesyncDetected {
+                                        frame: *local_frame,
+                                        local_checksum: *local_checksum,
+                                        remote_checksum: *remote_checksum,
+                                        addr: remote.peer_addr(),
+                                    });
+                                    // exit function desync already detected
+                                    return;
```

Comment:
If there are more than 2 players, this will only report desyncs for one of them... (and having that info is probably useful in order to figure out which player is the odd one out)

Maybe move the two inner loops here to a new helper function? Would also make it possible to unit test which is nice if performance optimizations produces ugly code.